edb/server/protocol/auth_ext/webauthn.py

```diff
@@ -284,8 +284,17 @@ async def create_authentication_options_for_email(
             cached_globally=True,
         )
         result_json = json.loads(result.decode())
+        if len(result_json) == 0:
+            raise errors.WebAuthnAuthenticationFailed(
+                "No WebAuthn credentials found for the email."
+            )
+
         user_handles: set[str] = {x["user_handle"] for x in result_json}
-        assert len(user_handles) == 1
+        if len(user_handles) > 1:
+            raise errors.WebAuthnAuthenticationFailed(
+                "Multiple user handles found for the email."
+            )
```

Comment:
Could still be an assert. It's really more of an ISE if there's multiple user handles, since the db should enforce that.